rest/rest-cli/src/test/java/functionaltests/NodeSourceCommandsFunctTest.java

```diff
@@ -130,95 +150,137 @@ private void defineNodeSourceAndListNodeSources(String nodeSourceName, boolean n
+    }
 
-        System.out.println(testLogString + " Test deployns command");
+    private void deployNodeSourceAndCheckOutput(String nodeSourceName, int expectedNumberOfNodes) {
+        String out;
+        System.out.println(LOG_HEADER + " Test deployns command");
 
         this.clearAndTypeLine("deployns( '" + nodeSourceName + "')");
         this.runCli();
-        this.waitForNodeSourceStatusToChange();
 
         out = this.capturedOutput.toString();
         System.setOut(this.stdOut);
         System.out.println(out);
 
         assertThat(out).contains("Node source successfully deployed.");
 
-        this.checkOccurrencesOfNodeSourceInNodeSourceList(testLogString, nodeSourceName);
-        this.checkOccurrencesOfNodeSourceInNodesList(testLogString, nodeSourceName);
+        this.checkOccurrencesOfNodeSourceInNodeSourceList(nodeSourceName);
+        this.checkOccurrencesOfNodeSourceInNodesList(nodeSourceName, expectedNumberOfNodes);
+    }
+
+    private void undeployNodeSourceAndCheckOutput(String nodeSourceName, boolean preemptUndeploy) {
 
-        System.out.println(testLogString + "Test undeployns command");
+        System.out.println(LOG_HEADER + "Test undeployns command");
 
         if (preemptUndeploy) {
             this.clearAndTypeLine("undeployns( '" + nodeSourceName + "', 'true')");
         } else {
             this.clearAndTypeLine("undeployns( '" + nodeSourceName + "')");
         }
         this.runCli();
-        this.waitForNodeSourceStatusToChange();
 
-        out = this.capturedOutput.toString();
+        String out = this.capturedOutput.toString();
         System.setOut(this.stdOut);
         System.out.println(out);
 
         assertThat(out).contains("Node source successfully undeployed.");
 
-        this.checkOccurrencesOfNodeSourceInNodeSourceList(testLogString, nodeSourceName);
-        this.checkNoOccurrenceOfNodeSourceInNodesList(testLogString, nodeSourceName);
+        this.checkOccurrencesOfNodeSourceInNodeSourceList(nodeSourceName);
+        this.checkNoOccurrenceOfNodeSourceInNodesList(nodeSourceName);
     }
 
-    private void checkOccurrencesOfNodeSourceInNodeSourceList(String testLogString, String nodeSourceName) {
-        System.out.println(testLogString + " List node sources");
+    private void editNodeSourceAndCheckOutput(String nodeSourceName) {
+
+        String nodeSourceInfrastructureClass = "org.ow2.proactive.resourcemanager.nodesource.infrastructure.LocalInfrastructure";
+        String nodeSourcePolicyClass = "org.ow2.proactive.resourcemanager.nodesource.policy.StaticPolicy";
+
+        System.out.println(LOG_HEADER + " Test editns command");
+
+        this.clearAndTypeLine("editns( '" + nodeSourceName + "', ['" + nodeSourceInfrastructureClass + "', '" +
+                              rmCredentialPath + "', " + MODIFIED_NB_NODES + ", 60000, ''], ['" +
+                              nodeSourcePolicyClass + "', 'ALL', 'ALL'], 'false')");
```

Comment:
This is not so easy to read.
Maybe you can use String.format?

Reply:
I created methods to generate the infrastructure string and the policy string.